src/sidebars/favourites/geometry-service.js

```diff
@@ -0,0 +1,63 @@
+(function() {
+    'use strict';
+
+    angular.module('openfin.geometry', [])
+        .factory('geometryService', [function() {
+
+            var Point = function(aX, aY) {
+                this.x = aX || 0;
+                this.y = aY || 0;
+            };
+            Point.constructor = Point;
+
+            var Rectangle = function(rect) {
+                this.origin = new Point(rect.left, rect.top);
+                this.extent = new Point(rect.width, rect.height);
+            };
+            Rectangle.constructor = Rectangle;
+
+            Rectangle.prototype = {
```

Comment:
We may want Chris or Colin to review / have a look? Unless you are 100% sure that this is legit.